### PR TITLE
chore: add v1 proxy tests and mocks for Marketo

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## What are the changes introduced in this PR?

Add v1 proxy tests for Marketo
New scenarios added = 3

## What is the related Linear task?

Resolves INT-1488

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3116/test-report.html

File: test/integrations/destinations/marketo/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        id: 1328328,
+        userId: 'gabi_userId_45',
+      },
+    ],
+    lookupField: 'id',
+  },
+  params,
+};
+
+export const testScenariosForV1API: ProxyV1TestData[] = [
```
Comment (utsabc) [inline test/integrations/destinations/marketo/dataDelivery/business.ts:73]: For Marketo there multiple response parsing scenarios where custom status code like 6XX and 7XX are parsed and application level errors are determined that is major use case for business scenarios we should test

File: test/integrations/destinations/marketo/dataDelivery/business.ts [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+          },
+        },
+      },
+    },
+  },
+  {
+    id: 'marketo_v1_scenario_5',
+    name: 'marketo',
+    description: '[Proxy v1 API] :: Test for DNS lookup failed scenario',
+    successCriteria: 'Should return a 400 status code with empty response',
```
Comment (utsabc) [inline test/integrations/destinations/marketo/dataDelivery/business.ts:263]: We already have these tests in `other.ts` if not lets add these there

File: test/integrations/destinations/marketo/network.ts [hunk 3]
```diff
@@ -970,3 +970,1081 @@ export const networkCallsData = [
...
+            userId: 'gabi_userId_45',
+          },
+        ],
+        lookupField: 'id',
+      },
+      headers: {
+        Authorization: 'Bearer test_token_2',
+        'Content-Type': 'application/json',
+        'User-Agent': 'RudderLabs',
+      },
```
Comment (utsabc) [inline test/integrations/destinations/marketo/network.ts:1039]: This is repeating

File: test/integrations/destinations/marketo/network.ts [hunk 4]
```diff
@@ -970,3 +970,1081 @@ export const networkCallsData = [
...
+        input: [
+          {
+            City: 'Tokyo',
+            Country: 'JP',
+            Email: 'gabi29@gmail.com',
+            PostalCode: '100-0001',
+            Title: 'Owner',
+            id: 1328328,
+            userId: 'gabi_userId_45',
+          },
```
Comment (utsabc) [inline test/integrations/destinations/marketo/network.ts:1031]: repeating

**Review by utsabc — CHANGES_REQUESTED** [review]

File: test/integrations/destinations/marketo/network.ts (hunk 4 above)
Comment (yashasvibajpai) [reply test/integrations/destinations/marketo/network.ts:1031]: Addressed

**Review by utsabc — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**